napari_aicssegmentation/_tests/view/workflow_steps_view_test.py

```diff
@@ -9,21 +8,20 @@
 class TestWorkflowStepsView:
     def setup_method(self):
         model = SegmenterModel()
-        # TODO mock this?
         model.active_workflow = WorkflowEngine().get_executable_workflow("sec61b", np.ones((1, 1, 1)))
         self._mock_controller: MagicMock = create_autospec(IWorkflowStepsController)
         self._view = WorkflowStepsView(self._mock_controller)
         self._view.load(model)
 
-    # TODO: Add real tests once we make the UI functional
-    def test_load_model(self):
-        pass
+    def test_show_workflow_diagram(self):
+        assert not self._view.window_workflow_diagram.isVisible()
+        self._view.btn_workflow_info.clicked.emit(False)
```

Comment:
Oh interesting, I take it you have to add `.emit(False)` just to indicate something was clicked?

Reply:
Yes, you use emit to programatically trigger the event. It's the closest to actually physically clicking on the button. Basically, i'm sure if we look at the code of the `QPushButton`, it will listen to mouse click events and run `self.clicked.emit()` when there's a click. So we just trigger the event manually instead, but from the perspective of our code, it has the same effect and most importantly any event handlers that we have connected to the button's `clicked` event will be triggered. That way we can test that our event handlers are properly connected.